src/node.cc

```diff
@@ -1110,6 +1111,57 @@ bool ShouldAbortOnUncaughtException(Isolate* isolate) {
 }
 
 
+void PromiseHook(PromiseHookType type,
+                 Local<Promise> promise,
+                 Local<Value> parent) {
```

Comment:
Don't have much context about domains but, AFAICS, the promise nor the parent is passed to the domain callback. Is that ok?